src/units.cpp

```diff
@@ -409,7 +409,8 @@ size_t Units::unitCount() const
 
 double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
 {
-    if ((units1 != nullptr) && (units2 != nullptr)) {
+    if ((units1 != nullptr) && (units2 != nullptr)
+        && (units1->unitCount() != 0) && (units2->unitCount() != 0)) {
```

Comment:
I have added two tests to show why this condition doesn't work for me.

Reply:
Yep, makes sense. Just added changes which pass both of those tests.